In [6]:
import nest_asyncio
nest_asyncio.apply()

In [1]:
from llama_cloud_services import LlamaParse
import os

In [2]:
LLAMA_API_KEY = os.environ.get("LLAMA_API_KEY")
if not LlamaParse:
    raise Exception("API key not found.")

In [3]:
parser = LlamaParse(
    api_key=LLAMA_API_KEY,
    num_workers=4,       # if multiple files passed, split in `num_workers` API calls
    verbose=True,
    language="en",       # optionally define a language, default=en
)

In [8]:
result = parser.parse("./test.pdf")

Started parsing the file under job_id 21c907c6-8b09-4d85-b44d-fd0e7f2f699e


/workspaces/gen-ai/env/lib/python3.12/site-packages/anyio/abc/_sockets.py:67: RuntimeWarning: coroutine 'LlamaParse.aparse' was never awaited
  SocketAttribute.raw_socket: lambda: self._raw_socket,


In [13]:
# get the llama-index markdown documents
markdown_documents = result.get_markdown_documents(split_by_page=True)
markdown_documents

[Document(id_='72d1fcd0-70e9-495c-95d2-d57dc32ad9e1', embedding=None, metadata={'page_number': 1, 'file_name': './test.pdf'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={}, metadata_template='{key}: {value}', metadata_separator='\n', text_resource=MediaResource(embeddings=None, data=None, text="# Print Claim UnitedHealthcare\n\n# Livwell Healthcare\n\n# Current Status\n\nClaim Processed\n\nMedical Claim # EV26375473 In-Network\n\nPatient Name: Daniel Rodrigues\n\nDate of service: 12/10/2024\n\nDate Received: 12/16/2024\n\n# Claim Payment Breakdown\n\n|Total Amount Billed|$594.00|Processed 12/18/2024|\n|---|---|---|\n|Plan Discount|$417.40|Processed 12/18/2024|\n|Plan Paid|$173.60|Processed 12/18/2024|\n|Your Total Amount Owed|$3.00|Not Marked As Paid|\n\n# Claim Provided Services\n\n|Provided Service|Amount Billed|Plan Discount|Plan Paid|Copay|Coinsurance|Deductible|Non-Covered|\n|---|---|---|---|---|---|---|---|\n|12/10/2024|$20.25|$17.25|$0.00|$0.00

In [18]:
markdown_documents[0].text_resource.text

"# Print Claim UnitedHealthcare\n\n# Livwell Healthcare\n\n# Current Status\n\nClaim Processed\n\nMedical Claim # EV26375473 In-Network\n\nPatient Name: Daniel Rodrigues\n\nDate of service: 12/10/2024\n\nDate Received: 12/16/2024\n\n# Claim Payment Breakdown\n\n|Total Amount Billed|$594.00|Processed 12/18/2024|\n|---|---|---|\n|Plan Discount|$417.40|Processed 12/18/2024|\n|Plan Paid|$173.60|Processed 12/18/2024|\n|Your Total Amount Owed|$3.00|Not Marked As Paid|\n\n# Claim Provided Services\n\n|Provided Service|Amount Billed|Plan Discount|Plan Paid|Copay|Coinsurance|Deductible|Non-Covered|\n|---|---|---|---|---|---|---|---|\n|12/10/2024|$20.25|$17.25|$0.00|$0.00|$0.00|$3.00|$0.00|\n|12/10/2024|$573.75|$400.15|$173.60|$0.00|$0.00|$0.00|$0.00|\n\n# Service Description\n\nRoutine procedure for taking a blood sample from a vein.\n\nComprehensive, preventive medical assessment for an adult (40-64 years old): This typically includes age- and gender-appropriate history, exam, counseling; educ

# Print Claim UnitedHealthcare

# Livwell Healthcare

# Current Status

Claim Processed

Medical Claim # EV26375473 In-Network

Patient Name: Daniel Rodrigues

Date of service: 12/10/2024

Date Received: 12/16/2024

# Claim Payment Breakdown

|Total Amount Billed|$594.00|Processed 12/18/2024|
|---|---|---|
|Plan Discount|$417.40|Processed 12/18/2024|
|Plan Paid|$173.60|Processed 12/18/2024|
|Your Total Amount Owed|$3.00|Not Marked As Paid|

# Claim Provided Services

|Provided Service|Amount Billed|Plan Discount|Plan Paid|Copay|Coinsurance|Deductible|Non-Covered|
|---|---|---|---|---|---|---|---|
|12/10/2024|$20.25|$17.25|$0.00|$0.00|$0.00|$3.00|$0.00|
|12/10/2024|$573.75|$400.15|$173.60|$0.00|$0.00|$0.00|$0.00|

# Service Description

Routine procedure for taking a blood sample from a vein.

Comprehensive, preventive medical assessment for an adult (40-64 years old): This typically includes age- and gender-appropriate history, exam, counseling; education and necessary lab work or imaging.

# Claim Codes

You have received a discount for using a health care professional in your plan's network: The total amount you owe may include your cost share (deductible, copay and coinsurance) and any non-covered amount after you meet your benefit limit for a covered service. (PPOOO8)

Benefits for this service have been applied to your deductible: The amount you owe shown on this statement is the amount you may owe your health care professional. (DEDOO3)

You have received a discount for using a health care professional in your plan's network. The total amount you owe may include your cost share (deductible, copay and coinsurance) and any non-covered amount after you meet your benefit limit for a covered service: (PPO0O8)

Claim Details

In [14]:
# get the llama-index text documents
text_documents = result.get_text_documents(split_by_page=False)
text_documents

[Document(id_='6a373bde-f6df-4ee8-9cfe-fe768877566e', embedding=None, metadata={'file_name': './test.pdf'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={}, metadata_template='{key}: {value}', metadata_separator='\n', text_resource=MediaResource(embeddings=None, data=None, text="12/20/24,8.28 PM                                Print Claim UnitedHealthcare\n  Livwell Healthcare                             Current Status\n  Solutions                                     Claim Processed\n   Medical Claim # EV26375473 In-Network            Submitted Under Review Processed\n  Patient Name: Daniel Rodrigues\n  Date of service:  12/10/2024\n  Date Received: 12/16/2024\n  Claim Payment Breakdown\n  Total Amount Billed                       $594.00                      Processed   12/18/2024\n  Plan Discount                             S417.40                      Processed   12/18/2024\n  Plan Paid                                 S173.60                      Proc

In [15]:
# get the image documents
image_documents = result.get_image_documents(
    include_screenshot_images=True,
    include_object_images=False,
    # Optional: download the images to a directory
    # (default is to return the image bytes in ImageDocument objects)
    image_download_dir="./images",
)
image_documents

[]

In [16]:
# Items will vary based on the parser configuration
for page in result.pages:
    print(page.text)
    print(page.md)
    print(page.images)
    print(page.layout)
    print(page.structuredData)

12/20/24,8.28 PM                                Print Claim UnitedHealthcare
  Livwell Healthcare                             Current Status
  Solutions                                     Claim Processed
   Medical Claim # EV26375473 In-Network            Submitted Under Review Processed
  Patient Name: Daniel Rodrigues
  Date of service:  12/10/2024
  Date Received: 12/16/2024
  Claim Payment Breakdown
  Total Amount Billed                       $594.00                      Processed   12/18/2024
  Plan Discount                             S417.40                      Processed   12/18/2024
  Plan Paid                                 S173.60                      Processed   12/18/2024
  Your Total Amount Owed $3.00                                               Not Marked As Paid
  Claim Provided Services
  Provided Service          Amount   Plan      Plan Paid Copay    Coinsurance Deductible Non-
                            Billed   Discount                                         Co